py_io_capture/io_capture.py

```diff
@@ -174,10 +199,41 @@ def process_args(orig_func, *args, **kwargs):
             except RecursionError:
                 record_arg = PythonReportError.RECURSION_LIMIT_EXCEEDED
             sys.setrecursionlimit(org_recursion_limit)
+
+        if is_class_instance(record_arg):
+            record_arg = obj2constructor(arg)
         processed[args_names[i]] = record_arg
 
     return processed
 
 
 def is_property(name):
-    return name.startswith("__") and name.endswith("__")
+    whitelist = ["__init__", "__setattr__"]
+    return name.startswith("__") and name.endswith("__") and name not in whitelist
+
+
+def is_class_instance(value: str) -> bool:
+    """check if a value is a class instance in Python.
+    We consider only a object <class_name object at 0xsome_address> as class instance
+
+    Args:
+        value (str): a reported value string
+
+    Returns:
+        bool: True if the value is an instance of some class, False otherwise
+    """
+    pattern = r"<([^}]*) object at 0x([0-9a-fA-F]{12})>"
+    match = re.match(pattern, value)
+    return bool(match)
+
+
+def obj2constructor(obj: Any) -> str:
+    if id(obj) in instance_tracker:
+        class_record = instance_tracker[id(obj)]
+        class_name = class_record[0]
+        args = class_record[1]
+        obj_str = f"{class_name}({', '.join(args)})"
```

Comment:
When converting to a constructor call, if arg is a string, then this concat needs to add the quotation.
For example, now is `LineNumber(A)`, need to be `LineNumber('A')`